### ISSUE TITLE
Element-space covariance accuracy is limited by cond(J), which grows with time from epoch

### ISSUE THREAD

**Issue by chrisfinlay** [chrisfinlay/sgp4jax#3]
## Summary

Covariance transforms through element space (`cov_teme_to_elements`, `cov_ric_to_elements`, and the `elements7` variants) have an accuracy floor of roughly `eps * cond(J) * ||Σ||`, where `J = elements_jacobian(satrec, jd, fr)`. `cond(J)` is not a fixed property of the code — it spans **five orders of magnitude** across satellites and grows with time since TLE epoch, so the same call is accurate to 1e-12 in one case and 1e-7 in another with nothing in the API to signal the difference.

This has already surfaced twice as CI flakiness rather than as a documented limitation:

- 19fc733 loosened `test_round_trip_ric_to_elements_to_ric` from `atol=1e-8` to `1e-7`.
- #2 had to make the same test's tolerance scale-aware: the identical computation lands at 9.40e-08 on macOS/jax 0.9.0.1 (94% of the 1e-7 budget) and 1.108e-07 on the Linux CI 3.10 runner with jax 0.6.2 (111%), so which platform runs the suite decided whether it was green. The round-trip matrix is bitwise identical across those commits — only the platform differed.

## Evidence

`cond(J)` and the measured round-trip error `max |RIC → elements → RIC − original|`, for `Σ_ric = diag([0.01, 1.0, 0.01, 1e-6, 1e-4, 1e-6])`:

| case | cond(J) | eps·cond·‖Σ‖ | inv (current) | solve | col-equilibrated | pinv |
|---|---|---|---|---|---|---|
| ISS, at epoch | 3.62e+07 | 8.03e-09 | 1.52e-08 | 4.91e-08 | 4.13e-08 | 5.65e-09 |
| ISS, +0.25 d | 1.16e+09 | 2.58e-07 | 2.55e-08 | 1.29e-08 | 1.87e-08 | 2.25e-08 |
| ISS, +1 d | 4.88e+09 | 1.08e-06 | 8.27e-09 | 4.61e-08 | 2.45e-08 | 1.48e-08 |
| Vanguard, at epoch | 1.37e+05 | 3.04e-11 | 9.54e-13 | 7.61e-13 | 4.84e-13 | 5.61e-13 |
| Vanguard, +0.25 d | 2.85e+07 | 6.33e-09 | 6.38e-11 | 2.42e-11 | 5.00e-11 | 1.26e-10 |
| Vanguard, +1 d | 5.10e+08 | 1.13e-07 | 2.22e-09 | 4.88e-10 | 1.14e-09 | 2.32e-09 |

The observed error tracks `eps * cond(J) * ||Σ||` to within about an order of magnitude, across a 4e4 range of conditioning.

## Why cond(J) is large and grows

The six elements are incommensurate — angles in radians (~1), `ecco` (~4e-4), `no_kozai` in rad/min (~0.067) — and `∂r/∂no_kozai` grows linearly with elapsed time. Column norms of `J` for the ISS:

| Δt from epoch | cond(J) | ‖∂r/∂inclo‖ | ‖∂r/∂nodeo‖ | ‖∂r/∂ecco‖ | ‖∂r/∂argpo‖ | ‖∂r/∂mo‖ | ‖∂r/∂no_kozai‖ |
|---|---|---|---|---|---|---|---|
| 0 d | 3.62e+07 | 6.31e+03 | 4.67e+03 | 8.69e+03 | 6.80e+03 | 6.79e+03 | 6.71e+04 |
| 1 d | 4.88e+09 | 6.42e+03 | 4.56e+03 | 8.36e+03 | 6.79e+03 | 6.80e+03 | 9.79e+06 |
| 7 d | 2.95e+10 | 6.79e+03 | 4.22e+03 | 6.92e+03 | 6.79e+03 | 6.80e+03 | 6.85e+07 |

The `no_kozai` column grows by three orders of magnitude over a week while the others are static. Since the transform is two-sided (`Σ_el = J⁻¹ Σ J⁻ᵀ`), the map's own conditioning goes as `cond(J)²` — ~2.5e17 for the ISS at +0.25 d, which is the figure quoted in 19fc733.

## What does *not* fix it

Swapping the linear-algebra routine is not the answer. Per the table, `solve`, `pinv`, and column equilibration land within a factor of ~5 of the current `jnp.linalg.inv`, sometimes better and sometimes worse — the error is dominated by the conditioning of `J` itself (and the accuracy of its autodiff entries), not by how it is inverted. Column equilibration does cut `cond(J)` from 4.98e+08 to 1.05e+07 (~47x) for the ISS at +0.25 d, but that does not show up as a consistent improvement in the round trip.

## Proposed fixes

1. **Expose the conditioning** (cheap, no behaviour change). Add something like `elements_jacobian_condition(satrec, jd, fr)`, document the accuracy floor `≈ eps * cond(J) * ||Σ||`, and note in the docstrings of the element-space transforms that accuracy degrades with time from epoch. Optionally warn above a threshold (say `cond(J) > 1e10`).
2. **Offer normalised element units.** Let the user work in dimensionless elements scaled by characteristic magnitudes so that `J` is well-scaled by construction, with the scaling exposed as part of the API rather than hidden. This attacks the cause rather than the symptom.
3. **Column-equilibrate internally** (`J = Js @ diag(d)`, invert `Js`, unscale). Reduces `cond(J)` ~47x in the ISS case. Should be benchmarked over a wider set of orbits and epochs before adopting — the measurements above do not yet justify it on round-trip accuracy alone.
4. **Test against the floor, not a constant.** Assert `error <= k * eps * cond(J) * ||Σ||` with a small `k`, so the tests encode the real numerical contract and stop needing to be loosened whenever a platform or JAX version shifts. #2 did half of this by scaling the tolerance to `||Σ||`; tying it to `cond(J)` as well would finish the job, and would turn a genuine conditioning regression into a failure instead of hiding it under a bigger constant.

## Reproduction

```python
import jax; jax.config.update("jax_enable_x64", True)
import jax.numpy as jnp, numpy as np
from sgp4jax import tle_to_satrec, elements_jacobian, cov_ric_to_teme, cov_teme_to_ric
from sgp4jax._kepler import _kepler_rv_teme

L1 = "1 25544U 98067A   20045.18587073  .00000950  00000-0  25302-4 0  9990"
L2 = "2 25544  51.6443 242.0161 0004397 264.6060 207.3845 15.49165514212791"
sat = tle_to_satrec(L1, L2)
cov_ric = np.diag([0.01, 1.0, 0.01, 1e-6, 1e-4, 1e-6])

for dt in (0.0, 0.25, 1.0):
    jd = jnp.float64(float(sat.jdsatepoch))
    fr = jnp.float64(float(sat.jdsatepochF) + dt)
    J = np.asarray(elements_jacobian(sat, jd, fr))
    r, v = _kepler_rv_teme(sat.inclo, sat.nodeo, sat.ecco, sat.argpo, sat.mo,
                           sat.no_kozai, sat.mu, sat.jdsatepoch, sat.jdsatepochF, jd, fr)
    cov_teme = np.asarray(cov_ric_to_teme(jnp.asarray(cov_ric), r, v))
    Ji = np.linalg.inv(J)
    back = np.asarray(cov_teme_to_ric(jnp.asarray(J @ (Ji @ cov_teme @ Ji.T) @ J.T), r, v))
    print(f"dt={dt:4.2f} d  cond(J)={np.linalg.cond(J):.2e}  "
          f"round-trip err={np.max(np.abs(back - cov_ric)):.2e}")
```

Measured with jax 0.9.0.1 / numpy 2.4.2 on macOS arm64.
